packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts

```diff
@@ -50,68 +85,74 @@ const builder = new SchemaBuilder({ scope: "treefuzz", libraries: [leaf.library]
  * During the fuzz test, {@link SchemaChange} can be generated which extends the allowed node types (with the node type being a generated uuid)
  * for each of our fields in our tree's current schema.
  */
-export const fuzzNode = builder.object("node", {
-	optionalChild: FlexFieldSchema.create(FieldKinds.optional, [Any]),
-	requiredChild: FlexFieldSchema.create(FieldKinds.required, [Any]),
-	sequenceChildren: FlexFieldSchema.create(FieldKinds.sequence, [Any]),
-});
-
-export type FuzzNodeSchema = typeof fuzzNode;
-
-export type FuzzNode = FlexTreeObjectNodeTyped<FuzzNodeSchema>;
-
-export const initialFuzzSchema = createTreeViewSchema([], leaf.library);
-
-export const fuzzSchema = builder.intoSchema(fuzzNode.objectNodeFieldsObject.optionalChild);
-
-export function createFuzzNode(
-	nodeTypes: LeafNodeSchema[],
-	schemaBuilder: SchemaBuilder,
-): typeof fuzzNode {
-	const node = schemaBuilder.objectRecursive("node", {
-		requiredChild: FlexFieldSchema.createUnsafe(FieldKinds.required, [
-			() => node,
-			leaf.number,
-			leaf.string,
-			leaf.handle,
-			...nodeTypes,
-		]),
-		optionalChild: FlexFieldSchema.createUnsafe(FieldKinds.optional, [
-			() => node,
-			leaf.number,
-			leaf.string,
-			leaf.handle,
-			...nodeTypes,
-		]),
-		sequenceChildren: FlexFieldSchema.createUnsafe(FieldKinds.sequence, [
-			() => node,
-			leaf.number,
-			leaf.string,
-			leaf.handle,
+export class FuzzNode extends builder.objectRecursive("node", {
+	optionalChild: builder.optionalRecursive(initialAllowedTypes),
+	requiredChild: builder.requiredRecursive(initialAllowedTypes),
+	sequenceChildren: SequenceChildren,
+}) {}
+type _checkFuzzNode = ValidateRecursiveSchema<typeof FuzzNode>;
+
+export type FuzzNodeSchema = typeof FuzzNode;
+
+export const initialFuzzSchema = createTreeViewSchema([]);
+export const fuzzFieldSchema = FuzzNode.info.optionalChild;
+
+export function createFuzzNodeSchema(
```

Comment:
what is this function for? why is there a second schema? can we doc it?

Reply:
This function is only called in `createTreeViewSchema`, so it shouldn't be called multiple times with the same schemaFactory. `createTreeViewSchema` makes its own instance of `SchemaFactory`, so it should be okay. I have removed the `export` from `createFuzzNodeSchema`, and added doc comments for both `createFuzzNodeSchema` and `createTreeViewSchema`